### PR TITLE
r128gain: 1.0.1 -> 1.0.3

### PR DESCRIPTION
**Pull request by AluisioASG**
###### Motivation for this change

Aside from the update, override the updateScript set by buildPythonApplication as it doesn't support GitHub sources. Manual intervention might still be needed for the patch file.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [x] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/audio/r128gain/ffmpeg-location.patch [hunk 1]
```diff
@@ -14,7 +12,7 @@ index 53fc3ef..f144e15 100755
 @@ -156,7 +156,7 @@ def get_r128_loudness(audio_filepaths, *, calc_peak=True, enable_ffmpeg_threadin
                                       os.devnull,
                                       **additional_ffmpeg_args,
-                                      f="null"),
+                                      f="null").global_args("-hide_banner", "-nostats"),
```
Comment (timokau) [inline pkgs/applications/audio/r128gain/ffmpeg-location.patch:15]: Does the `-hide_banner` and `-nostats` belong here? Seems unrelated to the ffmpeg change, and if its not nixos-functionality related the patch probably doesn't belong here. Maybe its something to propose upstream?

**Review by timokau — CHANGES_REQUESTED** [review]

File: pkgs/applications/audio/r128gain/ffmpeg-location.patch (hunk 1 above)
Comment (AluisioASG) [reply pkgs/applications/audio/r128gain/ffmpeg-location.patch:15]: That's just the patch context, that line was changed upstream and I had to update it so the patch applied again.

Comment (timokau) [reply pkgs/applications/audio/r128gain/ffmpeg-location.patch:15]: Oh I was confused by the diff of a diff. My bad.

**Review by timokau — APPROVED** [review]
